client/settings/express-checkout-settings/payment-request-button-preview.js

```diff
@@ -101,87 +149,78 @@ const PaymentRequestButtonPreview = () => {
 			}
 			setIsLoading( false );
 		} );
-	}, [ stripe, setPaymentRequest, setIsLoading ] );
+	}, [ stripe, setPaymentRequest, setIsLoading, isStripeEceEnabled ] );
 
 	/**
 	 * If stripe is loading, then display nothing.
 	 * If stripe finished loading but payment request button failed to load (null), display info section.
 	 * If stripe finished loading and payment request button loads, display the button.
 	 */
 
-	return (
-		<>
-			{ ( isWooPayEnabled ||
-				( isPaymentRequestEnabled && paymentRequest ) ) && (
-				<div
-					className="payment-method-settings__preview"
-					data-theme={ theme }
-				>
-					{ isWooPayEnabled && (
-						<WoopayExpressCheckoutButton
-							isPreview={ true }
-							buttonSettings={ {
-								type: buttonType,
-								text: 'Buy',
-								theme: theme,
-								height: `${
-									buttonSizeToPxMap[ size ] ||
-									buttonSizeToPxMap.medium
-								}px`,
-								size,
-							} }
-						/>
-					) }
-					{ isPaymentRequestEnabled &&
-						! isLoading &&
-						paymentRequest && (
-							<PaymentRequestButtonElement
-								key={ `${ buttonType }-${ theme }-${ size }` }
-								onClick={ ( e ) => {
-									e.preventDefault();
-								} }
-								options={ {
-									paymentRequest: paymentRequest,
-									style: {
-										paymentRequestButton: {
-											type: buttonType,
-											theme: theme,
-											height: `${
-												buttonSizeToPxMap[ size ] ||
-												buttonSizeToPxMap.medium
-											}px`,
-										},
-									},
-								} }
-							/>
-						) }
-				</div>
-			) }
-			{ ! isWooPayEnabled && ! isPaymentRequestEnabled && (
-				<InlineNotice icon status="info" isDismissible={ false }>
-					{ __(
-						'To preview the express checkout buttons, ' +
-							'activate at least one express checkout.',
-						'woocommerce-payments'
-					) }
-				</InlineNotice>
-			) }
-			{ isPaymentRequestSettingsPage() &&
-				isPaymentRequestEnabled &&
-				! isLoading &&
-				! paymentRequest && (
-					<InlineNotice icon status="info" isDismissible={ false }>
-						{ __(
-							'To preview the Apple Pay and Google Pay buttons, ' +
-								'ensure your device is configured to accept Apple Pay or Google Pay, ' +
-								'and view this page using the Safari or Chrome browsers.',
-							'woocommerce-payments'
-						) }
-					</InlineNotice>
```

Comment:
I refactored this whole section to make it easier to read. No more ternary-hell and instead there are several if-statements I find much easier to read. It might be easier to look at the changes here in a split diff — or even locally in your editor — rather than trying to parse this with the default view.